src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -3441,19 +3455,75 @@ impl Coordinator {
                     // the columns refer only to referenced subsources. The best solution to
                     // this is to ensure that we simply don't try to generate the table cast
                     // in the first place.
-                    curr_references.contains(&name)
-                });
+                    if curr_subsources_by_reference.contains_key(&name) {
+                        curr_names_by_oid.insert(table.oid, name.clone());
+                        curr_tables_by_name.insert(name, table);
+                    }
+                }
 
                 mz_ore::soft_assert_eq_or_log!(
-                    curr_details.tables.len(),
-                    curr_references.len(),
+                    curr_tables_by_name.len(),
+                    curr_subsources_by_reference.len(),
                     "PostgresSourcePublicationDetails must have entry for every reference"
                 );
 
-                let referenced_oids: BTreeSet<_> =
-                    curr_details.tables.iter().map(|t| t.oid).collect();
+                // Track the set of text columns.
+                let mut curr_text_columns_mapping: BTreeMap<u32, BTreeSet<u16>> = BTreeMap::new();
+
+                // Drop all text columns that are not currently referred to.
+                curr_text_columns.retain(|column_qualified_reference| {
+                    mz_ore::soft_assert_eq_or_log!(
```

Comment:
Why is it safe to log but continue if this fails in release mode?

Reply:
It's annoying to return an error during `retain`. I'll move these checks outside of retain and throw internal errors.